client/src/plus/updates/api.ts

```diff
@@ -35,9 +37,51 @@ interface Page {
   last: number;
 }
 
-export function useUpdates(page: number) {
+function composeUrl({
+  isAuthenticated,
+  searchParams,
+}: {
+  isAuthenticated: boolean;
+  searchParams: URLSearchParams;
+}): string {
+  let url = "/api/v2/updates/";
+  let params = new URLSearchParams();
+
+  for (const [key, value] of searchParams.entries()) {
+    switch (key) {
+      case "page":
+        params.set(key, value);
+        break;
+
+      case "show":
+        if (isAuthenticated) {
+          url += "watched/";
```

Comment:
maybe a comment here to explain?

Reply:
Resolved via 6019f8f71379f9bfe1ff7d943600a9de667c9284.